subsys/bluetooth/host/l2cap.c

```diff
@@ -1256,6 +1252,25 @@ static void l2cap_chan_seg_sent(struct bt_conn *conn, void *user_data)
 	l2cap_chan_tx_resume(BT_L2CAP_LE_CHAN(chan));
 }
 
+static bool test_and_dec(atomic_t *target)
+{
+	static struct k_spinlock lock;
+	k_spinlock_key_t key;
+
+	key = k_spin_lock(&lock);
+
+	if (!(*target)) {
```

Comment:
I realize the spin lock is sufficient, but this currently only works because atomic_t is typedef'd to unsigned int.
The abstraction will not break in future if we use atomic_get, atomic_dec (below).